### PR TITLE
refactor: move default sorting filter logic to store

### PR DESCRIPTION
**Pull request by AlfonsoGhislieri**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Pushes the logic for the default sorting values for the research and howto list pages into the stores .

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/stores/Howto/howto.store.tsx [hunk 1]
```diff
@@ -48,15 +48,22 @@ export class HowtoStore extends ModuleStore {
...
   public allHowtos: IHowtoDB[]
+
   @observable
   public selectedCategory: string
+
   @observable
   public searchValue: string
 
+  @observable
+  public activeSorter: string
```
Comment (evakill) [inline src/stores/Howto/howto.store.tsx:62]: can this be `ItemSortingOption` to have better type safety?

File: src/stores/Howto/howto.store.tsx [hunk 2]
```diff
@@ -70,21 +77,26 @@ export class HowtoStore extends ModuleStore {
...
+    this.allDocs$.subscribe((docs: IHowtoDB[]) => {
+      this.filterSorterDecorator = new FilterSorterDecorator<any>(docs)
+      // Sets default starting sort filter for howto list items
+      this.updateActiveSorter('Created')
+    })
   }
 
   public updateActiveSorter(query: string) {
     this.allHowtos = this.filterSorterDecorator?.sort(query)
+    this.activeSorter = query
```
Comment (evakill) [inline src/stores/Howto/howto.store.tsx:99]: i know this isn't in your diff, but `query` is kinda a weird name for this, can we call it `sorter`?

**Review by evakill — COMMENTED** [review]
the FilterSorterDecorater tests look solid - might also be good to add a test of the default sort for howto and filter stores? up to you

File: src/stores/Howto/howto.store.tsx (hunk 1 above)
Comment (AlfonsoGhislieri) [reply src/stores/Howto/howto.store.tsx]: Yes absolutely, great suggestion added in e108bd

File: src/stores/Howto/howto.store.tsx (hunk 2 above)
Comment (AlfonsoGhislieri) [reply src/stores/Howto/howto.store.tsx]: e108bdd changes this

**Pull request merged**